react/Minicart.tsx

```diff
@@ -0,0 +1,82 @@
+import React, { FC } from 'react'
+import { useOrderForm } from 'vtex.order-manager/OrderForm'
+import { useCssHandles } from 'vtex.css-handles'
+
+import MinicartIconButton from './components/MinicartIconButton'
+import DrawerMode from './components/DrawerMode'
+
+import { MinicartContextProvider, useMinicartState } from './MinicartContext'
+import PopupMode from './components/Popup'
+
+interface Props {
+  variation: 'popup' | 'drawer' | 'link'
```

Comment:
I did not create an `enum` but I created a dedicated `type` with the options still as `strings`. I prefer that since the values we receive via props are of type `string`

Reply:
You still get the same type-checking and auto-complete :)